### ISSUE TITLE
traefik 1.x not compatible with kubernetes 1.22

### ISSUE THREAD

**Issue by r0bj** [traefik/traefik#8343]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

There are some API removals in kubernetes 1.22 (https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/#api-changes):
- all beta Ingress APIs (the extensions/v1beta1 and networking.k8s.io/v1beta1 API versions)

Running traefik 1.7.30 (latest from 1.x branch) in kubernetes 1.22 (released as stable) generates an error:
```
E0807 19:25:26.977338       1 reflector.go:205] github.com/containous/traefik/vendor/k8s.io/client-go/informers/factory.go:86: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
```

According to information here: https://traefik.io/blog/traefik-2-1-in-the-wild traefik 1.x branch is supported until the end of 2021.

Migration traefik from 1.x to 2.x across whole organization with multiple teams is quite a long journey so supporting stable API `networking.k8s.io/v1` by traefik 1.x would make it easier. Especially with cloud kubernetes offerings like GKE where kubernetes upgrades to new version are automatic - in this case using traefik 1.x (still supported) will cause not easy to recover disaster.

### What did you see instead?

```
E0807 19:25:26.977338       1 reflector.go:205] github.com/containous/traefik/vendor/k8s.io/client-go/informers/factory.go:86: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
```

### What version of Traefik are you using?

```
Digest: sha256:1fa5d3b0e00c6b2b10eab9ea0e149f0930f36fb46dee9f56ec2a2c04570685f6
Status: Downloaded newer image for traefik:v1.7.30-alpine
Version:      v1.7.30
Codename:     maroilles
Go version:   go1.16.3
Built:        2021-04-08_02:50:29PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration?

```yaml
# (paste your configuration here)
```

Add more configuration information here.

**Comment by n0nvme** [comment]
same problem with traefik v2.4.13 and kubernetes 1.22

**Comment by ddtmachado** [comment]
Hello @r0bj,

Thanks for your feedback.

For now, we do not plan to introduce compatibility with K8s 1.22 in Traefik V1. Let me explain why:

* Yes, you are right. Traefik V1 will be supported until the end of 2021 but so will K8s 1.21. And as this version works and is pretty new, there is still a good alternative.
* Introducing compatibility for 1.22+ requires a good amount of effort because of legacy code. Even if we would start right now, it would take time to get it done and thus, lowering the time "left" it will be supported.

Our suggestion for now is to use the remaining time to upgrade to v2, rather then trying to use V1 with K8s 1.22.

That being said, as usual, we are all ears if we are wrong. We just don't see a lot of adoption for Traefik V1 in Kube 1.22. If we suddenly see it, yet again, we are open discuss.

As this is primarily a question, I'll close the issue as we try to keep the open issues clear from questions.

If we missed something, please feel free to re-open it.

@n0nvme for V2 this will be fixed on 2.5, which is coming really soon.

**Comment by n0nvme** [comment]
@ddtmachado thanks for your answer!
I've already found that the current release candidate for 2.5 already supports k8s 1.22 and switched to it. Works fine, but we are still waiting for a stable 2.5 release to update our production cluster to v1.22
